conan/tools/cmake/cmakedeps/templates/target_configuration.py

```diff
@@ -128,6 +129,11 @@ def template(self):
                      PROPERTY INTERFACE_COMPILE_OPTIONS
                      $<$<CONFIG:{{configuration}}>:${{'{'}}{{pkg_name}}_COMPILE_OPTIONS{{config_suffix}}}> APPEND)
 
+        # This is only used for '#pragma comment(lib, "foo")' (automatic link)
+        set_property(TARGET {{root_target_name}}
+                     PROPERTY INTERFACE_LINK_DIRECTORIES
```

Comment:
Now this is interesting. New from 3.13, maybe this can avoid us all the dirty stuff we are doing in our macro to locate the actual library files inside the ``lib_paths``? 

Apparently it is discouraged from the CMake docs (https://cmake.org/cmake/help/latest/command/target_link_directories.html#command:target_link_directories):
> Note This command is rarely necessary and should be avoided where there are other choices. Prefer to pass full absolute paths to libraries where possible, since this ensures the correct library will always be linked. The [find_library()](https://cmake.org/cmake/help/latest/command/find_library.html#command:find_library) command provides the full path, which can generally be used directly in calls to [target_link_libraries()](https://cmake.org/cmake/help/latest/command/target_link_libraries.html#command:target_link_libraries). Situations where a library search path may be needed include:

So yeah, maybe we shouldn't do it.

Maybe a case for the traits? If the trait is ``libs=True``, but cpp_info.libs is empty, then add this?

Reply:
Anyway I would limit this to Windows until a legit use case exists in others.

Reply:
Even if this is only for Windows, I can see some risks adding this line in Windows (recall Windows is one of the OS that struggle with long compilation lines too). 

The thing is that this is borderline bad practice, linking with pragma is not clean, kind of a very messy feature. Better make it very opt-in. Possible options are:

- Explicit opt-in in the CMakeDeps customer side
- Explicit opt-in as a ``cpp_info.property`` in the package side
- Implicit opt-in with libs=empty + libdirs=not-empty combination defined in package side

I think I would add anything unless an opt-in enables this.